### PR TITLE
fix(HeaderBar): fix proptypes warnings

### PR DESCRIPTION
**Pull request by vyshkov**
**What is the problem this PR is trying to solve?**

A lot of warnings:

```
preview.bundle.js:1104 Warning: Failed prop type: The prop `logo.t` is marked as required in `HeaderBar`, but its value is `undefined`.
    in HeaderBar (created by Translate(HeaderBar))
    in Translate(HeaderBar)
    in div (created by Story)
    in div (created by Story)
    in Story
    in div
    in I18nextProvider
```

```
preview.bundle.js:1104 Warning: Failed prop type: The prop `help.t` is marked as required in `HeaderBar`, but its value is `undefined`.
    in HeaderBar (created by Translate(HeaderBar))
    in Translate(HeaderBar)
    in div (created by Story)
    in div (created by Story)
    in Story
    in div
    in I18nextProvider
```

```
Warning: Failed prop type: The prop `products.t` is marked as required in `HeaderBar`, but its value is `undefined`.
    in HeaderBar (created by Translate(HeaderBar))
    in Translate(HeaderBar)
    in div (created by Story)
    in div (created by Story)
    in Story
    in div
    in I18nextProvider
```

```
Warning: Failed prop type: The prop `renderers.name` is marked as required in `User`, but its value is `undefined`.
    in User (created by HeaderBar)
    in HeaderBar (created by Translate(HeaderBar))
    in Translate(HeaderBar)
    in div (created by Story)
    in div (created by Story)
    in Story
    in div
    in I18nextProvider
```

**What is the chosen solution to this problem?**

Fixed poptypes definitions

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/HeaderBar/HeaderBar.component.js [hunk 1]
```diff
@@ -271,13 +269,13 @@ if (process.env.NODE_ENV !== 'production') {
 	};
 
 	HeaderBar.propTypes = {
-		logo: PropTypes.shape(Logo.propTypes),
+		logo: PropTypes.shape({ ...Logo.propTypes, t: undefined }),
```
Comment (jare-talend) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: I find weird to add t: undefined. _.omit(object, 't') will be better ?

Comment (vyshkov) [reply packages/components/src/HeaderBar/HeaderBar.component.js]: @jare-talend but we don't have lodash dependency in components ?

Comment (vyshkov) [reply packages/components/src/HeaderBar/HeaderBar.component.js]: Ah, i see it.

File: packages/components/src/HeaderBar/HeaderBar.component.js [hunk 2]
```diff
@@ -255,12 +256,10 @@ if (process.env.NODE_ENV !== 'production') {
 	};
 
 	User.propTypes = {
-		renderers: PropTypes.shape({
-			ActionDropdown: PropTypes.func,
-			name: PropTypes.string.isRequired,
-			firstName: PropTypes.string,
-			lastName: PropTypes.string,
-		}),
+		ActionDropdown: PropTypes.func,
```
Comment (jsomsanith) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: `ActionDropdown` is in a `renderers` object

**Pull request merged**